File: acquire/src/main/java/hu/elte/alkfejl/acquire/model/Ad.java
package hu.elte.alkfejl.acquire.model;

import hu.elte.alkfejl.acquire.model.post.NewAd;
import java.text.SimpleDateFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@Entity
@Table(name = "ADS")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Ad extends BaseEntity {
   
    
    public Ad(NewAd ad,User costumer){
        name = ad.getName();
        description = ad.getDescription();
        location = ad.getLocation();
        price = ad.getPrice();
        deadline = ad.getDeadline();
        costumer_id = costumer;
        deliver_id = costumer;
        status = status.PENDING;
    }
    
    @Column(nullable = false)
    private String name;
    
    
    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private int price;

    @Column(nullable = false)
    private Date deadline;

    
    @JoinColumn(name = "COSTUMER_ID")
    @ManyToOne(fetch = FetchType.EAGER)
    private User costumer_id;
    
    
    @JoinColumn(name = "DELIVER_ID")
    @ManyToOne(fetch = FetchType.EAGER)
    private User deliver_id;

    @Enumerated(EnumType.STRING)
    @Column
    private Status status;

    public enum Status{
        PENDING,ACCEPTED,DONE,UNRATED
    }

    public void clone(NewAd ad){
        this.description=ad.getDescription();
        this.location=ad.getLocation();
        this.deadline=ad.getDeadline();
        this.price=ad.getPrice();
    }
    
//    public String getDeadline(){
//        SimpleDateFormat ft = 
//                new SimpleDateFormat ("E yyyy.MM.dd hh:mm:ss ");
//        return ft.format(deadline);
//    }
//    
//    public Long getCostumer_id(){
//        return costumer_id.getId();
//    }
//    public Long getDeliver_id(){
//        return deliver_id.getId();
//    }
}

File: social-delivery/src/app/classes/ad.ts
export enum Status{
    PENDING = "PENDING",
    ACCEPTED = "ACCEPTED",
    DONE = "DONE",
    UNRATED = "UNRATED"


}

export class Ad {
    private _id:number;
    private _status:Status;
    private _costumer_id:number;
    private _deliver_id:number;

    public constructor(
        private _name:string,
        private _description:string,
        private _location:string,
        private _price:number,
        private _deadline:number,
    ) {}
    get deliver_id():number{
        return this._deliver_id;
    }

    get costumer_id():number{
        return this._costumer_id;
    }

    get id():number{
        return this._id;
    }

    get name():string{
        return this._name;
    }
    
    get description(): string {
        return this._description;
    }
    set description(value: string) {
        this._description = value;
    }
    get location():string{
        return this._location;
    }
    set location(value:string){
        this._location = value;
    }
    get price():number{
        return this._price;
    }
    set price(value:number){
        this._price = value;
    }
    get deadline():number{
        return this._deadline;
    }
    set deadline(value:number){
        this._deadline = value;
    }
    get status():Status{
        return this._status;
    }
    set status(value:Status){
        this._status = value;
    }
}
